sync/src/block/extension.rs

```diff
@@ -759,14 +756,12 @@ impl Extension {
             }
             State::SnapshotChunk(..) => {}
             State::Full => {
-                let (mut completed, pivot_score_changed) = if let Some(peer) = self.header_downloaders.get_mut(from) {
-                    let before_pivot_score = peer.pivot_score();
+                let (mut completed, is_lagging) = if let Some(peer) = self.header_downloaders.get_mut(from) {
                     let encoded: Vec<_> = headers.iter().map(|h| EncodedHeader::new(h.rlp_bytes().to_vec())).collect();
                     peer.import_headers(&encoded);
-                    let after_pivot_score = peer.pivot_score();
-                    (peer.downloaded(), before_pivot_score != after_pivot_score)
+                    (peer.downloaded(), peer.is_lagging())
```

Comment:
What do you think about a variable like `pivot_hash_changed`. Then the behavior is the same as before.

Reply:
I updated this part to use `is_caught` method, which compares the pivot with the best hash.